src/main/groovy/com/github/mgk/gradle/S3Plugin.groovy

```diff
@@ -33,9 +34,14 @@ abstract class S3Task extends DefaultTask {
     @Input
     String bucket
 
+    static AmazonS3Client s3Client
+
     String getBucket() { bucket ?: project.s3.bucket }
 
     def getS3Client() {
+        if (s3Client) {
```

Comment:
I have run into issues reusing the `s3Client` object (I seem to recall it expiring or becoming invalid after a certain period of time) so I create it for each operation.

If you really want s3Client reuse, please put that in a separate PR and make it configurable via an environment variable.